### PR TITLE
Add bank example

### PR DESCRIPTION
**Pull request by janmasrovira**
This PR adds the Juvix program for the `Bank` example provided in the [Leo
  workshop repository](https://github.com/AleoHQ/workshop/blob/master/basic_bank/src/main.leo).

### CODE REVIEW

File: examples/milestone/Bank/Bank.juvix [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+--- Runs an ;IO; action on the chain.
+axiom runOnChain : {B : Type} -> IO -> B -> B;
+
+--- Computes the hash associated with an ;Address;.
+axiom hashAddress : Address -> Field;
+
+--- Applies a function n times.
+iterate : {A : Type} -> Nat -> (A -> A) -> A -> A;
+iterate zero _ a := a;
+iterate (suc n) f a := iterate n f (f a);
```
Comment (jonaprieto) [inline examples/milestone/Bank/Bank.juvix:89]: What if we actually add this function here: [Stdlib](https://github.com/anoma/juvix-stdlib/tree/main/Stdlib)
/Function.juvix

Comment (janmasrovira) [reply examples/milestone/Bank/Bank.juvix:89]: I've created a pr for the stdlib: https://github.com/anoma/juvix-stdlib/pull/52

File: examples/milestone/Bank/juvix.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+dependencies:
+- .juvix-build/stdlib/
+name: bank
+version: 0.0.0
```
[the remark this answers is not in the document]
Comment (janmasrovira) [reply examples/milestone/Bank/juvix.yaml:4]: the `description` field is not part of `juvix.yaml`

[the remark this answers is not in the document]
Comment (janmasrovira) [reply examples/milestone/Bank/juvix.yaml:4]: I can remove the readme, since I repeated the same info in the judoc comment of the top module

**Review by jonaprieto — APPROVED** [review]

**Pull request merged**